### ISSUE TITLE
childView and emptyView no longer support `Backbone.View`s, now they require `Marionette.View`s in 2.4.2

### ISSUE THREAD

**Issue by nomcopter** [marionettejs/backbone.marionette#2651]
This commit: https://github.com/marionettejs/backbone.marionette/commit/2cefc20d9ef2552a398b8b051b3151107f25a9fd

Has introduced a bug in which `childView` and `emptyView` in `CollectionView`s can no longer take a `Backbone.View` and require a `Marionette.View` instead. This is because of the added call to `view._getNestedViews()` in `addEmptyView` and `_addChildView`, which `Backbone.View` does not define.

**Comment by samccone** [comment]
great catch @nomcopter !

**Comment by ianmstew** [comment]
Wow, great catch.  Handling this now.

**Comment by megawac** [comment]
One way we can address this is through something like the `classify` changes being proposed, we can do something like `staticify`, in which we move all  methods on a classes prototype to its static properties (ala python classes).

This would let us to simply change calls of `view._getNestedViews()` (and any other Marionette specific calls) to `Marionette.View._getNestedViews(view)` without worrying about whether its a BB View or a Marionette View

I think this would be a simple and forward-compatible way to do this, Thoughts?

**Comment by megawac** [comment]
As for implementation it can be done quite easily like

```js
// Move functions on the prototype to static class properties
// to allow a user to call prototype methods ala python.
// Example: `Marionette.View.bindEntityEvents(myView, events);`
Marionette.staticify = function(klass) {
  var proto = _.omit(klass.prototype);
  _.chain(proto).functions()
   .each(function(key) {
    klass[key] = function(instance) {
      return proto[key].apply(instance, _.rest(arguments));
    };
   });
};

Marionette.staticify(Marionette.View)
```

**Comment by samccone** [comment]
So I am -1 to modifying the bb prototype, the reason being that it simply does not scale. 

Marionette works off of the idea that if you provide a view that adheres to the interface API that what we expect (see a backbone view or Mn View) you can just use it inside of Marionette and things "work". 
Thus making things interop with BB without any prototype shenanigans means things like ampersand's view and other view layers can be dropped into Mn and things just work. Without having to `classify` or `metalify

**Comment by megawac** [comment]
@samccone the whole idea of the staticify thing I was proposing was to avoid augmenting any `BB` prototypes. It would let you do `Marionette.Model.destroy(normalBackboneModel)` or `Marionette.ItemView.bindEntity(normalBackboneView, events)`

**Comment by samccone** [comment]
I see and do the work inside of the method to "normalize" the object?

**Comment by paulfalgout** [comment]
@samccone the issue not to classify Bb is here: https://github.com/marionettejs/backbone.marionette/issues/2637

**Comment by ianmstew** [comment]
Just a heads up, but this issue's PR is now updated.  Conversation should continue there.  See #2658.

**Comment by ianmstew** [comment]
Fixed by #2658.
